server/aesdsocket.c

```diff
@@ -138,25 +140,59 @@ static void *handle_client(void *arg) {
 
 if (pkt && len) {
     size_t start = 0;
+    int did_seek_io = 0;
     while (start < len) {
         size_t i = start;
         while (i < len && pkt[i] != '\n') i++;
 
         if (i < len && pkt[i] == '\n') {
             size_t linelen = (i + 1) - start;  // include '\n'
-            pthread_mutex_lock(&file_mutex);
-            append_packet(DATAFILE, pkt + start, linelen);
-            pthread_mutex_unlock(&file_mutex);
-            start = i + 1;  // next line
+#if USE_AESD_CHAR_DEVICE
+if (linelen >= 18 && strncmp(pkt + start, "AESDCHAR_IOCSEEKTO", 18) == 0) {
+unsigned X = 0, Y = 0;
+    int parsed = sscanf(pkt + start, "AESDCHAR_IOCSEEKTO:%u,%u", &X, &Y);
+    if (parsed == 2) {
+        struct aesd_seekto st = { .write_cmd = X, .write_cmd_offset = Y };
+
+        pthread_mutex_lock(&file_mutex);                  // like your friend: serialize device ops
```

Comment:
Shouldn't need locking here since you have locking in your device implementation.